### ISSUE TITLE
Make the video and window sizes fit

### ISSUE THREAD

**Issue by Aurnytoraink** [Rafostar/clapper#218]
I feel it annoying when I try to resize the window to make it matches the video size.
Furthermore, resizing the corner should keep the video ratio (like photo editor) but if the user wants to resize freely, he cans use `shift + move` combination.

I found [this](https://matrix-client.matrix.org/_matrix/media/r0/download/gnome.org/9bb17342d5440ebdf83e00095e286a47ee4ab914) on the GTK matrix's server, which may help you to understand my idea

**Comment by Aurnytoraink** [comment]
Sorry #30

**Comment by Rafostar** [comment]
@Aurnytoraink 

Yes, a duplicate, but the app in the video you attached (that can auto-resize itself to contents preferred size) seems useful. Is there any code of that app available anywhere for reference?

**Comment by Aurnytoraink** [comment]
There is no code ATM, but I found the [original post](https://mastodon.xyz/web/statuses/107780379660613814)

According to the [conversation](https://matrix.to/#/!aUhETchlgthwWVQzhi:matrix.org/$16446648192167ohUST:gnome.org?via=gnome.org&via=matrix.org&via=mozilla.org) on the Matrix server, he seems to use this [GDK function](https://docs.gtk.org/gdk4/struct.ToplevelSize.html)
